### PR TITLE
Feat/sqs notifications

### PR DESCRIPTION
**Pull request by bpalazzi512**
# Add basic notification system and AWS infrastructure as code
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/66)
Added a notification system integrated with AWS SQS to fire off notifications. Any notification sent will get sent via an AWS Lambda, which is triggered from the SQS queue. Within the application, notifications that should happen immediately (i.e. thank you for registering, etc), will get sent right to the SQS queue. The queue processes events as soon as they come in. I built a simple Go app (in /infra/modules/main/lambda) to actually send the notifications. For scheduled notifications, they get added to a table in the DB. Every 5 minutes a cron job running in the background scans the table for notifications that are due to be sent. When it identifies the notifications, it sends them to the SQS queue to be processed. 

Alongside all of that, I also put all of our infra as terraform in the `/infra` directory. Here you can see the different environments. I have the buckets in their own separate env, with `dev` housing the infra that is on there now. You can easily make a prod env by essentially copying and pasting what's in dev and calling the main module. 

I also added a mechanism for running aws infra locally using a tool called localstack. You can call `make up-localstack` to bring everything up with localstack (alongside toggling the variable in the local.env)

# How Has This Been Tested?
I manually tested to see that email and push notifications worked. For email, I hardcoded my email as the "to" email (as resend can only send emails to one account under testing mode) and verified that I got an email upon hitting the registration endpoint. For push notifications, I had to create a mock app in expo and spun up the frontend on my phone to get my expo push code. Once that happened, I added it as as scheduled notification in the db. This tested that push notifications and scheduled notifications do indeed work. Tested it on both aws and localstack

## Screenshots
This is a screenshot I took of me receiving both notifications at the same time
IMG_0877

# Checklist

## General
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [x] I have written unit tests for my code
- [x] I have written integration tests where appropriate
- [x] New and existing unit tests pass locally with my changes
- [x] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [x] I have tested error handling and edge cases
- [x] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]

### Android
[Add Android screenshots here]

### CODE REVIEW

**Comment by adescoteaux1** [comment]
Did you happen to look into if the push notifs will work the same on android/if theres anything that will need to be configued differently?

File: backend/internal/service/handler/registration/handler.go [hunk 1]
```diff
@@ -11,17 +12,24 @@ type Handler struct {
...
-	StripeClient              stripeClient.StripeClientInterface
+	StripeClient 			  stripeClient.StripeClientInterface
+	NotificationService       *notification.Service
 }
 
+// func NewHandler(registrationRepo storage.RegistrationRepository, childRepo storage.ChildRepository, guardianRepo storage.GuardianRepository, eventOccurrenceRepo storage.EventOccurrenceRepository, notifService *notification.Service) *Handler {
+// 	OrganizationRepository    storage.OrganizationRepository
+// 	StripeClient              stripeClient.StripeClientInterface
+// }
+
```
Comment (adescoteaux1) [inline backend/internal/service/handler/registration/handler.go]: delete old code

File: backend/internal/notification/scheduler.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+			slog.Error("Failed to process notification", "id", notification.ID, "error", err)
+			// Update status to failed
+			_, updateErr := s.repo.Notification.UpdateNotificationStatus(ctx, notification.ID, models.NotificationStatusFailed)
+			if updateErr != nil {
+				slog.Error("Failed to update notification status to failed", "id", notification.ID, "error", updateErr)
+			}
+			continue
+		}
+
+		// Update status to sent (even though it's just queued, we mark it as sent to prevent reprocessing)
```
Comment (josh-torre) [inline backend/internal/notification/scheduler.go]: I would add a quick comment to the definitions of the status enum then to make it clear that "sent" means sent to AWS for processing

File: backend/jobs/sendScheduledNotificationsJob.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"skillspark/internal/models"
+	"log"
+)
+
+func (j *JobScheduler) SendScheduledNotificationsJob() {
```
[the remark this answers is not in the document]
Comment (bpalazzi512) [reply backend/jobs/sendScheduledNotificationsJob.go]: just deleted it, was in my old scheduler that wasn't used
